### PR TITLE
Adding more tests

### PR DESCRIPTION
**Pull request by mayhem**
This adds tests for core functionality. At this point the most important modules/features seem covered -- we should discuss what other modules/functions need more tests.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -18,7 +18,7 @@
         "License :: OSI Approved :: MIT License",
         "Operating System :: OS Independent",
     ],
-    python_requires='>=3.7',
+    python_requires='>=3.6',
```
Comment (alastair) [inline setup.py:21]: This has gone down from 3.8 to 3.7 to 3.6. I'm just curious to know why - is there a reason that you need an older version, or are you just interested in supporting all currently supported pythons?

File: troi/musicbrainz/tests/test_related_artist_credits.py [hunk 2]
```diff
@@ -1,68 +1,55 @@
...
-import troi.musicbrainz.recording_lookup
+import troi.musicbrainz.related_artist_credits
 
-request_json = [
-                    { "[recording_mbid]": "a96bf3b6-651d-49f4-9a89-eee27cecc18e" }, 
-                    { "[recording_mbid]": "ec5b8aa9-7483-4791-a185-1f599a0cdc35" }
-               ]
-
-return_json = [
+return_json = """[
```
Comment (alastair) [inline troi/musicbrainz/tests/test_related_artist_credits.py:8]: this is a string but others in `test_recording_lookup.py` and `test_area_random_recordings.py` use an object and json.dumps when setting the mock's return value

File: troi/tests/test_utils.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+class TestPatches(unittest.TestCase):
+
+    def test_discover_patches(self):
+        patches = discover_patches()
+
+        assert len(patches) == 2
+        assert "daily-jams" in patches
+        assert "area-random-recordings" in patches
+
+        print(patches)
```
Comment (alastair) [inline troi/tests/test_utils.py:16]: extra print

File: troi/tests/test_utils.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+from troi.utils import discover_patches
+from troi.patch import Patch
+
+
+class TestPatches(unittest.TestCase):
+
+    def test_discover_patches(self):
+        patches = discover_patches()
+
+        assert len(patches) == 2
```
Comment (alastair) [inline troi/tests/test_utils.py:12]: are you expecting to have to change this every time we add a new built-in patch?

**Review by alastair — APPROVED** [review]
Tests look fine here 👍

File: setup.py (hunk 1 above)
Comment (mayhem) [reply setup.py:21]: My mac still has 3.6...

File: troi/tests/test_utils.py (hunk 4 above)
Comment (mayhem) [reply troi/tests/test_utils.py:12]: Yes. My intention was to be a bit more explicit about adding new patches, but I don't feel strongly about it.

File: setup.py (hunk 1 above)
Comment (alastair) [reply setup.py:21]: given that we're not focusing on building in docker, that's a perfectly fine reason.

File: troi/tests/test_utils.py (hunk 3 above)
Comment (mayhem) [reply troi/tests/test_utils.py:16]: Fixed and committed to main, I didn't see these comments at first.

File: troi/musicbrainz/tests/test_related_artist_credits.py (hunk 2 above)
Comment (mayhem) [reply troi/musicbrainz/tests/test_related_artist_credits.py:8]: Ah, yes. Fixed.

File: troi/tests/test_utils.py (hunk 3 above)
Comment (alastair) [reply troi/tests/test_utils.py:16]: not sure if you prefer that I mark reviews as changes requested even when there are only small changes to be made? I didn't think that it was worth doing 2 rounds for such a small change. Or I can be more explicit on the review message.

Comment (mayhem) [reply troi/tests/test_utils.py:16]: I think if you used "request changes" if there are issues that you really want to see fixed and "approve" if you're fine with merging, regardless of what comments you've left, I think that would work well for me. WDYT?

**Pull request merged**